### PR TITLE
Added Modal  Component

### PR DESCRIPTION
**Pull request by atharvabhanage02**
Added 
- Alert Modal
- Simple Modal
- Confirmation Modal

Here is the Preview Link : 
https://deploy-preview-13--apex-ui.netlify.app/documentation/docs

### CODE REVIEW

**Review by psychoactiv — APPROVED** [review]
Good to merge, do the below changes though

File: components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
+.modal-container{
+    margin-top: 1.87rem;
+    display: flex;
+    flex-direction: row;
+    gap: 3.12rem;
+    justify-content: center;
+    align-items: center;
+}
+.dialog-box{
+    background-color: white;
```
Comment (psychoactiv) [inline components/modal/modal.css]: Could use color variables

File: components/modal/modal.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    border: 2px solid #ccc;
+    border-radius: 10px;
+}
+.dialog-header{
+    font-size: 1.5rem;
+    font-weight: bolder;
+}
+.dialog-description{
+    margin: 1rem 0px;
+    color: grey;
```
Comment (psychoactiv) [inline components/modal/modal.css:28]: Color  variable

File: components/modal/modal.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+.dialog-actions{
+    display: flex;
+    justify-content: flex-end;
+}
+.modal-action-btn{
+    font-size: 1.2rem ;
+    font-weight: bolder;
+    background-color: white;
```
Comment (psychoactiv) [inline components/modal/modal.css]: Same as above

**Review by psychoactiv — APPROVED** [review]

File: components/modal/modal.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
+.modal-container{
+    margin-top: 1.87rem;
+    display: flex;
+    flex-direction: row;
```
Comment (ozairkhan2810) [inline components/modal/modal.css]: no need to mention. by default it is row

File: components/modal/modal.css [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    align-items: center;
+}
+.dialog-box{
+    background-color: white;
+    display: flex;
+    flex-direction: column;
+    gap: 1rem;
+    width: 15.6rem;
+    padding: 1.25rem;
+    border: 2px solid #ccc;
```
Comment (ozairkhan2810) [inline components/modal/modal.css]: use color variable

File: components/modal/modal.css [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+.modal-item{
+    padding: 10px;
+    font-size: 1.2rem;
+}
+.item{
+    font-size: 1.2rem;
+    padding: 10px;
+}
+.item:hover{
+    background-color: #ccc;
```
Comment (ozairkhan2810) [inline components/modal/modal.css]: use color variable

**Review by ozairkhan2810 — APPROVED** [review]
few suggestions, rest looks good to merge

File: components/modal/modal.css (hunk 4 above)
Comment (atharvabhanage02) [reply components/modal/modal.css]: I will remove that . Thanks for pointing that out .

File: components/modal/modal.css (hunk 6 above)
Comment (atharvabhanage02) [reply components/modal/modal.css]: Yess

**Pull request merged**